tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AdoNet/DbScopeFactory.cs

```diff
@@ -0,0 +1,118 @@
+// <copyright file="DbScopeFactory.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Data;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AdoNet
+{
+    internal static class DbScopeFactory
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(DbScopeFactory));
+
+        public static Scope CreateDbCommandScope(Tracer tracer, IDbCommand command)
+        {
+            if (TryGetIntegrationDetails(command.GetType().FullName, out var integrationId, out var dbTypeName))
+            {
+                var integration = IntegrationRegistry.GetIntegrationInfo(integrationId.ToString());
```

Comment:
This reminds me, I need to rebase my source generator branch once 2.0 is out, one of the source generators I have adds a fast `switch` expression alternative for `ToString()` for enums to avoid the reflection

Reply:
Looks like a lot of this is gone now that we dropped the string-based API.